edx_name_affirmation/tasks.py

```diff
@@ -75,32 +77,35 @@ def idv_update_verified_name_task(self, attempt_id, user_id, name_affirmation_st
             verified_name_obj.save()
 
         log.info(
-            'Updated VerifiedNames for user={user_id} with verification_attempt_id={attempt_id} to '
+            'Updated VerifiedNames for user={user_id} with platform_verification_attempt_id={attempt_id} to '
             'have status={status}'.format(
                 user_id=user_id,
                 attempt_id=attempt_id,
                 status=name_affirmation_status
             )
         )
-    else:
-        # otherwise if there are no entries, we want to create one.
-        user = User.objects.get(id=user_id)
-        verified_name = VerifiedName.objects.create(
-            user=user,
-            verified_name=photo_id_name,
-            profile_name=full_name,
-            verification_attempt_id=attempt_id,
-            status=name_affirmation_status,
-        )
-        log.error(
-            'Created VerifiedName for user={user_id} to have status={status} '
-            'and verification_attempt_id={attempt_id}, because no matching '
-            'attempt_id or verified_name were found.'.format(
-                user_id=user_id,
-                attempt_id=attempt_id,
-                status=verified_name.status
-            )
+
+        if updated_for_attempt_id or verified_name_qs:
```

Comment:
This actually fixes what appears to be an existing bug. If a user has a previously verified name using a different method (eg proctoring) and an new IDV creation event is emitted a new verified name isn't created. From a user standpoint this doesn't really happen since the UI will post a blank name before that event but it still seemed like a gap.